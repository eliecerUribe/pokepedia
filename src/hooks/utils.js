import axios from "axios";
import { pokemonType, getColorByOpacity } from "../utils";

export const transformData = (data) =>
  data.map(async (elem) => {
    const { name, url } = elem;
    const { data } = await axios.get(url);
    const avatar = data.sprites.other["official-artwork"].front_default;
    const [{ type }] = data.types;
    const backgroundColor = getColorByOpacity(
      pokemonType[type.name].color,
      0.85
    );

    return {
      name,
      id: data.id,
      avatar,
      types: data.types,
      typeColor: pokemonType[type.name].color,
      typeImage: pokemonType[type.name].img,
      backgroundColor,
    };
  });
