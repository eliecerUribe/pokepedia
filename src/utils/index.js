import figthing from "../images/fighting.png";
import fire from "../images/fire.png";
import hypchic from "../images/hypchic.png";
import leave from "../images/leave.png";
import lighting from "../images/lightning.png";
import poison from "../images/poison.png";
import rock from "../images/rock.png";
import water from "../images/water.png";
import wind from "../images/wind.png";
import bug from "../images/bug.png";
import ground from "../images/ground.png";
import ghost from "../images/ghost.png";
import steel from "../images/steel.png";
import ice from "../images/ice.png";
import dragon from "../images/dragon.png";
import fairy from "../images/fairy.png";

export const apiUrl = "https://pokeapi.co/api/v2/pokemon";

export const pokemonType = {
  normal: { color: "rgba(170, 166, 127, 1)", img: "" },
  fighting: {
    color: "rgba(193, 34, 57, 1)",
    img: figthing,
  },
  flying: { color: "rgba(168, 145, 236, 1)", img: wind },
  poison: { color: "rgba(164, 62, 158, 1)", img: poison },
  ground: { color: "rgba(222, 193, 107, 1)", img: ground },
  rock: { color: "rgba(182, 158, 49, 1)", img: rock },
  bug: { color: "rgba(167, 183, 35, 1)", img: bug },
  ghost: { color: "rgba(112, 85, 155, 1)", img: ghost },
  steel: { color: "rgba(183, 185, 208, 1)", img: steel },
  fire: { color: "rgba(245, 125, 49, 1)", img: fire },
  water: { color: "rgba(100, 147, 235, 1)", img: water },
  grass: { color: "rgba(116, 203, 72, 1)", img: leave },
  electric: { color: "rgba(249, 207, 48, 1)", img: lighting },
  psychic: { color: "rgba(251, 85, 132, 1)", img: hypchic },
  ice: { color: "rgba(154, 214, 223, 1)", img: ice },
  dragon: { color: "rgba(112, 55, 255, 1)", img: dragon },
  dark: { color: "rgba(117, 87, 76, 1)", img: "" },
  fairy: { color: "rgba(230, 158, 172, 1)", img: fairy },
  unknown: { color: "", img: "" },
  shadow: { color: "rgba(183, 185, 208, 1)", img: "" },
};

export const getColorByOpacity = (str, opacity) =>
  str.replace(
    /(rgba\(\d+,\s*\d+,\s*\d+,\s*)\d+(\.\d+)?(\s*\))/,
    `$1${opacity}$3`
  );
