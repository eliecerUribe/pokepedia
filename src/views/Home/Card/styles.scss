.card {
  align-items: center;
  border-radius: 16px;
  box-shadow: 0 0 0 transparent;
  display: flex;
  flex-direction: column;
  justify-content: center;
  padding: 14px 16px;
  position: relative;
  transition: box-shadow 0.2s ease;
  width: 226px;

  &:hover {
    box-shadow: 0px 10px 29px -10px rgba(0, 0, 0, 0.75);
    cursor: pointer;
  }

  &:hover .background-image {
    display: block;
  }

  .background-image {
    bottom: -45px;
    display: none;
    padding-right: 1rem;
    position: absolute;
    right: -61px;
  }

  .id {
    color: #fff;
    font-size: 20px;
    font-style: normal;
    font-weight: 700;
    line-height: 30px;
    text-align: left;
    width: 100%;
  }

  .name {
    align-items: center;
    color: #1d1d1d;
    display: flex;
    font-size: 22px;
    font-style: normal;
    font-weight: 800;
    justify-content: space-between;
    line-height: 18px;
    letter-spacing: 1px;
    position: relative;
    text-align: left;
    width: 100%;

    img {
      width: 26px;
      position: absolute;
      right: 0;
    }
  }
}
