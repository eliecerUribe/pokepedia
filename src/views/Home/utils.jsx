export const apiUrl = "https://pokeapi.co/api/v2";

export const pokemonType = {
  normal: "rgba(170, 166, 127, 1)",
  fighting: "rgba(193, 34, 57, 1)",
  flying: "rgba(168, 145, 236, 1)",
  poison: "rgba(164, 62, 158, 1)",
  ground: "rgba(222, 193, 107, 1)",
  rock: "rgba(182, 158, 49, 1)",
  bug: "rgba(167, 183, 35, 1)",
  ghost: "rgba(112, 85, 155, 1)",
  steel: "rgba(183, 185, 208, 1)",
  fire: "rgba(245, 125, 49, 1)",
  water: "rgba(100, 147, 235, 1)",
  grass: "rgba(116, 203, 72, 1)",
  electric: "rgba(249, 207, 48, 1)",
  psychic: "rgba(251, 85, 132, 1)",
  ice: "rgba(154, 214, 223, 1)",
  dragon: "rgba(112, 55, 255, 1)",
  dark: "rgba(117, 87, 76, 1)",
  fairy: "rgba(230, 158, 172, 1)",
  unknown: "",
  shadow: "",
};

export const getColorByOpacity = (str, opacity) =>
  str.replace(
    /(rgba\(\d+,\s*\d+,\s*\d+,\s*)\d+(\.\d+)?(\s*\))/,
    `$1${opacity}$3`
  );
