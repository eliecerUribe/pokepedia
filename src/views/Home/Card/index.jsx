import PropTypes from "prop-types";
import { getFormattedId } from "../../../utils";
import pokeball from "../../../images/pokeball1.svg";
import "./styles.scss";

const Card = ({ id, name, avatar, typeImage, backgroundColor, onClick }) => {
  const formattedId = getFormattedId(id);

  return (
    <div
      className="card"
      style={{ background: `${backgroundColor}` }}
      onClick={onClick}
    >
      <div className="name">
        {name.toUpperCase()}
        {typeImage && <img src={typeImage} alt={name} width="12px" />}
      </div>
      <div className="id">{formattedId}</div>
      <img src={avatar} alt={name} className="avatar" width="200" />
      <img src={pokeball} alt="pokeball" className="background-image" />
    </div>
  );
};

Card.propTypes = {
  id: PropTypes.number.isRequired,
  name: PropTypes.string.isRequired,
  avatar: PropTypes.string.isRequired,
  typeImage: PropTypes.string,
  backgroundColor: PropTypes.string,
  onClick: PropTypes.func.isRequired,
};

export default Card;
