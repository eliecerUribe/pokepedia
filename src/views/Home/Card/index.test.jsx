import { render, screen } from "@testing-library/react";
import { getFormattedId } from "../../../utils";
import Card from "./";

describe("Card", () => {
  const mockProps = {
    id: 27,
    name: "pikachu",
    avatar: "pikachu.png",
    typeImage: "electric.png",
    backgroundColor: "yellow",
    onClick: jest.fn(),
  };

  it("matches snapshot with required props", () => {
    const { container } = render(<Card {...mockProps} />);

    expect(container).toMatchSnapshot();
  });

  it("renders correctly with formatted id and name", () => {
    render(<Card {...mockProps} />);

    const formattedId = getFormattedId(mockProps.id);
    const formattedName = mockProps.name.toUpperCase();
    const pokemon = {
      id: screen.getByText(formattedId),
      name: screen.getByText(formattedName),
    };

    expect(pokemon.id).toBeInTheDocument();
    expect(pokemon.name).toBeInTheDocument();
  });
});
