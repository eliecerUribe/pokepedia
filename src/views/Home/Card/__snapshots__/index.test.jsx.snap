// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card matches snapshot with required props 1`] = `
<div>
  <div
    class="card"
    style="background: yellow;"
  >
    <div
      class="name"
    >
      PIKACHU
      <img
        alt="pikachu"
        src="electric.png"
        width="12px"
      />
    </div>
    <div
      class="id"
    >
      # 0027
    </div>
    <img
      alt="pikachu"
      class="avatar"
      src="pikachu.png"
      width="200"
    />
    <img
      alt="pokeball"
      class="background-image"
      src="pokeball1.svg"
    />
  </div>
</div>
`;
